tutorials/io/plot_30_reading_fnirs_data.py

```diff
@@ -59,18 +60,140 @@
 If triggers are sent using the ``digaux`` port of the recording hardware, MNE
 will also read the ``digaux`` data and create annotations for any triggers.
-
-
-Storing of optode locations
-===========================
-
-NIRs devices consist of light sources and light detectors.
-A channel is formed by source-detector pairs.
-MNE stores the location of the channels, sources, and detectors.
-
-
-.. warning:: Information about device light wavelength is stored in
-             channel names. Manual modification of channel names is not
-             recommended.
-
 """  # noqa:E501
+
+# sphinx_gallery_thumbnail_number = 2
+
+###############################################################################
+# Loading legacy data in csv or tsv format
+# ========================================
+#
+# Many legacy fNIRS measurements are stored in csv and tsv formats.
+# These formats are not officially supported in MNE as there is no
+# standardisation of the file format -
+# the naming and ordering of channels, the type and scaling of data, and
+# specification of sensor positions varies between each vendor.
+# Instead, we suggest that data is converted to the format approved by the
+# Society for functional near-infrared spectroscopy called
+# `SNIRF <https://github.com/fNIRS/snirf>`_,
+# they provide a number of tools to convert your legacy
+# data to the SNIRF format.
+# However, due to the prevalence of these legacy files we provide
+# a template example of how you may read data in t/csv formats.
+
+import numpy as np
+import pandas as pd
+import mne
+
+
+###############################################################################
+# First, we generate an example csv file which will then be loaded in to MNE.
+# This step would be skipped if you have actual data you wish to load.
+# We simulate 16 channels with 100 samples of data and save this to a file
+# called fnirs.csv.
+
+pd.DataFrame(np.random.normal(size=(16, 100))).to_csv("fnirs.csv")
+
+
+###############################################################################
+# Next, we will load the example csv file.
+# The metadata must be specified manually as the csv file does not contain
+# information about channel names, types, sample rate etc.
+
+data = pd.read_csv('fnirs.csv')
+
+# In MNE the naming of channels MUST follow this structure of
+# `S#_D# type` or `S#_D# wavelength`, where # is replaced
+# by the appropriate source and detector number, type is
+# either hbo or hbr, and wavelength is specified in nm.
+ch_names = ['D1_S1 hbo', 'D1_S1 hbr', 'D1_S2 hbo', 'D1_S2 hbr',
+            'D1_S3 hbo', 'D1_S3 hbr', 'D1_S4 hbo', 'D1_S4 hbr',
+            'D2_S5 hbo', 'D2_S5 hbr', 'D2_S6 hbo', 'D2_S6 hbr',
+            'D2_S7 hbo', 'D2_S7 hbr', 'D2_S8 hbo', 'D2_S8 hbr']
+
+ch_types = ['hbo', 'hbr', 'hbo', 'hbr',
+            'hbo', 'hbr', 'hbo', 'hbr',
+            'hbo', 'hbr', 'hbo', 'hbr',
+            'hbo', 'hbr', 'hbo', 'hbr']
```

Comment:
@rob-luke @larsoner @drammock, I'd like to add a file in mne/channels/data/montages/ that would contain the source detector mapping for fNIRS montages since we have them in the vendor provided file (https://github.com/mne-tools/mne-python/pull/9141/commits/e0f8cc64fbdd67e7169af324e6eb4af23099f657) and work on a function to load them.
Something like "optode-channels.txt" that would contain the mappings and wavelengths for each fNIRS montage.

The way I see it is like this:
```python
# Like this to get optical density channels
ch_names, ch_types = default_fnirs_channels('artinis-octamon', type='od')

# Or this to get concentration changes channels
ch_names, ch_types = default_fnirs_channels('artinis-octamon', type='hb')

# And then
info = mne.create_info(ch_names=ch_names, ch_types=ch_types, sfreq=sfreq)
raw = mne.io.RawArray(data, info)

raw.set_montage('artinis-octamon')
```
This will save user's time when loading the data before setting the montage.

Reply:
I don't think we should do this. My thoughts are:

1. Artinis is now exporting their data in SNIRF format so this will only help for legacy files.
2. Artinis are currently writing an MNE reader for their non snirf data that should be ready soon, that would make this suggestion redundant.
2. legacy datasets should be converted to snirf and then imported in to MNE. The society for fNIRS provides tools to convert data to snirf. If your dataset can't be converted using their tools then we should open an issue or PR fix there.
5. In this pr we provide an example of how to load legacy data, but we state that it's not supported due to the lack of standardisation in how people use csvs to store fNIRS data. If we start writing more functions to load these file formats then we will need to provide support for them. 
3. I don't want to encourage loading data in this way, as the community is trying to consolidate on the agreed snirf format.
4. It's extra code to maintain and doesn't really reduce the lines of code in a script much.

However, if everyone else wants this feature then I suggest it's implemented in another PR as I think it will have some complications (mainly setting the wavelength fields consistently), and this pr is nicely self contained as is.